pkg/daemon/daemon.go

```diff
@@ -632,18 +632,31 @@ func (dn *Daemon) runLoginMonitor(stopCh <-chan struct{}, exitCh chan<- error) {
 			case <-worker:
 				return
 			default:
+				if dn.node == nil {
+					continue
+				}
 				buf := make([]byte, 1024)
 				l, err := stdout.Read(buf)
 				if err != nil {
 					if err == io.EOF {
+						glog.Warning("runLoginMonitor got EOF while reading the journal, exiting")
 						return
 					}
 					exitCh <- err
 					return
 				}
 				if l > 0 {
-					line := strings.Split(string(buf), "\n")[0]
-					glog.Infof("Detected a new login session: %s", line)
+					lines := strings.Split(string(buf), "\n")
```

Comment:
Is iterating over the lines and checking for "New session" faster/more efficient than a regexp check on the buffer (Match with bytes or Find with string)?

Reply:
Simple example (again -- may not be better or worse)

```go

package main

import (
        "fmt"
        "regexp"
)

var RE = regexp.MustCompile(`New session`)


func newSession(d []byte) {
        fmt.Printf("%t: %s", RE.Match(d), d)
        fmt.Println("\n-")
}


func main() {
        newSession([]byte(`Starting Login Service...
Watching system buttons on /dev/input/event0 (Power Button)
New seat seat0.
New session 5 of user core.
Removed session 5.`))
        newSession([]byte(`Starting Login Service...
Watching system buttons on /dev/input/event0 (Power Button)
Removed session 5.`))
}
```

Output:
```
$ ./main 
true: Starting Login Service...
Watching system buttons on /dev/input/event0 (Power Button)
New seat seat0.
New session 5 of user core.
Removed session 5.
-
false: Starting Login Service...
Watching system buttons on /dev/input/event0 (Power Button)
Removed session 5.
-

```

Reply:
Right, uhm, I don’t really know, the regexp has to match “starting with that pattern” also. Not sure, I think we can tweak this with a regexp for sure, I don’t think this buffers grow enough to worry about the for loop over those lines - I think benchmarking this would be great to do, but we don’t have much data to run the benchmark against lol 
I don’t really have a preference tho, I’m ok changing to a regexp that works 👍

Reply:
Na, keep it as is -- the above can be an option to try if we see a performance hit.